newIDE/app/src/EventsFunctionsList/index.js

```diff
@@ -123,6 +133,54 @@ export default class EventsFunctionsList extends React.Component<Props, State> {
     );
   };
 
+  _getFunctionThumbnail = (eventsFunction: ?gdEventsFunction) => {
+    switch (eventsFunction.getFunctionType()) {
+      default:
+        return 'res/functions/function32.png';
+      case 0:
+        if (
+          eventsFunction.getName() === 'onSceneLoaded' ||
+          eventsFunction.getName() === 'onFirstSceneLoaded' ||
+          eventsFunction.getName() === 'onCreated'
+        ) {
+          return 'res/functions/create32.png';
+        }
+        if (
+          eventsFunction.getName() === 'onScenePreEvents' ||
+          eventsFunction.getName() === 'onScenePostEvents' ||
+          eventsFunction.getName() === 'doStepPreEvents' ||
+          eventsFunction.getName() === 'doStepPostEvents'
+        ) {
+          return 'res/functions/step32.png';
+        }
+        if (
+          eventsFunction.getName() === 'onScenePaused' ||
+          eventsFunction.getName() === 'onDeActivate'
+        ) {
+          return 'res/functions/deactivate32.png';
+        }
+        if (
+          eventsFunction.getName() === 'onSceneResumed' ||
+          eventsFunction.getName() === 'onActivate'
+        ) {
+          return 'res/functions/activate32.png';
+        }
+        if (
+          eventsFunction.getName() === 'onSceneUnloading' ||
+          eventsFunction.getName() === 'onDestroy'
+        ) {
+          return 'res/functions/destroy32.png';
+        } else {
+          return 'res/functions/action32.png';
+        }
+      case 1:
+        return 'res/functions/condition32.png';
+      case 2:
+        return 'res/functions/expression32.png';
+      case 3:
+        return 'res/functions/expression32.png';
```

Comment:
Don't use the numbers directly, that makes the code unreadable and hard to understand 😱 
```suggestion
      case gd.EventsFunction.Action:
        if (
          eventsFunction.getName() === 'onSceneLoaded' ||
          eventsFunction.getName() === 'onFirstSceneLoaded' ||
          eventsFunction.getName() === 'onCreated'
        ) {
          return 'res/functions/create32.png';
        }
        if (
          eventsFunction.getName() === 'onScenePreEvents' ||
          eventsFunction.getName() === 'onScenePostEvents' ||
          eventsFunction.getName() === 'doStepPreEvents' ||
          eventsFunction.getName() === 'doStepPostEvents'
        ) {
          return 'res/functions/step32.png';
        }
        if (
          eventsFunction.getName() === 'onScenePaused' ||
          eventsFunction.getName() === 'onDeActivate'
        ) {
          return 'res/functions/deactivate32.png';
        }
        if (
          eventsFunction.getName() === 'onSceneResumed' ||
          eventsFunction.getName() === 'onActivate'
        ) {
          return 'res/functions/activate32.png';
        }
        if (
          eventsFunction.getName() === 'onSceneUnloading' ||
          eventsFunction.getName() === 'onDestroy'
        ) {
          return 'res/functions/destroy32.png';
        } else {
          return 'res/functions/action32.png';
        }
      case gd.EventsFunction.Condition:
        return 'res/functions/condition32.png';
      case gd.EventsFunction.Expression:
        return 'res/functions/expression32.png';
      case gd.EventsFunction.StringExpression:
        return 'res/functions/expression32.png';
```

Reply:
Disregard this, Alex's proposition is better